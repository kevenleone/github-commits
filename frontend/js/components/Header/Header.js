import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav } from 'reactstrap';
import './Header.scss';

const Header = () => {
  const { user } = useSelector((state) => state.base);
  const [isOpen, setIsOpen] = useState(false);

  function callLogout() {
    // eslint-disable-next-line no-alert
    const shouldLogout = window.confirm(`${user.name}, you sure you want logout?`);
    if (shouldLogout) {
      window.location.href = '/logout';
    }
  }

  const toggle = () => setIsOpen(!isOpen);
  return (
    <div className="HeaderComponent">
      <Navbar color="dark" dark expand="md">
        <NavbarBrand href="/">GitHub Commits</NavbarBrand>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="mr-auto" navbar />
          <p className="user_name">
            <span className="name">Welcome, {user.name}</span> <br />
            <button className="logout" type="button" onClick={callLogout}>
              Logout
            </button>
          </p>
          <img alt="user" className="user_avatar" src={user.avatar_url} />
        </Collapse>
      </Navbar>
    </div>
  );
};

export default Header;
