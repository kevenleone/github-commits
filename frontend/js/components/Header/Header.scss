.HeaderComponent {
  img.user_avatar {
    width: 40px;
    height: 40px;
    border-radius: 20px;
    margin-left: 10px;
  }

  button.logout {
    margin-top: 2px;
    background: none;
    color: #fafafa;
    font-weight: bold;
  }

  p.user_name {
    color: #dadada;
    position: absolute;
    top: 30px;
    right: 70px;
  }
}
