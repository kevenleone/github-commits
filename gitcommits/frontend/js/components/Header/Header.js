import React, { useState } from 'react';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav } from 'reactstrap';

const Header = () => {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);

  return (
    <div>
      <Navbar color="dark" dark expand="md">
        <NavbarBrand href="/">GitHub Commits</NavbarBrand>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="mr-auto" navbar />
          <span>Welcome!</span>
        </Collapse>
      </Navbar>
    </div>
  );
};

export default Header;
