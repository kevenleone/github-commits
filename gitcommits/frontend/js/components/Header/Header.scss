.HeaderComponent {
  img.user_avatar {
    width: 40px;
    height: 40px;
    border-radius: 20px;
    margin-left: 10px;
  }

  span.user_name {
    color: #dadada;
  }
}
