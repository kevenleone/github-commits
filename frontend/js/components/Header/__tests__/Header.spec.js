import React from 'react';
import { Provider } from 'react-redux';
import sagaMiddleware from 'redux-saga';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import Header from '../Header';

import { Header as Index } from '..';

const mockStore = configureMockStore([sagaMiddleware]);

describe('Header', () => {
  const store = mockStore({
    base: { user: { name: 'Keven' } },
  });
  global.confirm = () => true;

  test('shoud render Header Index', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Index />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with success', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render and check user name', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    expect(wrapper.find('span.name').text()).toStrictEqual('Welcome, Keven');
  });

  test.skip('should render and click on logout', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    wrapper.find('button.logout').simulate('click');
    // jest.spyOn(window, 'confirm').mockImplementation(() => true);
  });
});
