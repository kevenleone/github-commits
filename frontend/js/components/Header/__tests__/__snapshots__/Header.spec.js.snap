// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header shoud render Header Index 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header>
    <div
      className="HeaderComponent"
    >
      <Navbar
        color="dark"
        dark={true}
        expand="md"
        tag="nav"
      >
        <nav
          className="navbar navbar-expand-md navbar-dark bg-dark"
        >
          <NavbarBrand
            href="/#/"
            tag="a"
          >
            <a
              className="navbar-brand"
              href="/#/"
            >
              GitHub Commits
            </a>
          </NavbarBrand>
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          >
            <button
              aria-label="Toggle navigation"
              className="navbar-toggler"
              onClick={[Function]}
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
          </NavbarToggler>
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Transition
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={350}
              unmountOnExit={false}
            >
              <div
                aria-expanded="false"
                className="collapse navbar-collapse"
                style={Object {}}
              >
                <Nav
                  className="mr-auto"
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <ul
                    className="mr-auto navbar-nav"
                  />
                </Nav>
                <p
                  className="user_name"
                >
                  <span
                    className="name"
                  >
                    Welcome, 
                    Keven
                  </span>
                   
                  <br />
                  <button
                    className="logout"
                    onClick={[Function]}
                    type="button"
                  >
                    Logout
                  </button>
                </p>
                <img
                  alt="user"
                  className="user_avatar"
                />
              </div>
            </Transition>
          </Collapse>
        </nav>
      </Navbar>
    </div>
  </Header>
</Provider>
`;

exports[`Header should render with success 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header>
    <div
      className="HeaderComponent"
    >
      <Navbar
        color="dark"
        dark={true}
        expand="md"
        tag="nav"
      >
        <nav
          className="navbar navbar-expand-md navbar-dark bg-dark"
        >
          <NavbarBrand
            href="/#/"
            tag="a"
          >
            <a
              className="navbar-brand"
              href="/#/"
            >
              GitHub Commits
            </a>
          </NavbarBrand>
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          >
            <button
              aria-label="Toggle navigation"
              className="navbar-toggler"
              onClick={[Function]}
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
          </NavbarToggler>
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Transition
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={350}
              unmountOnExit={false}
            >
              <div
                aria-expanded="false"
                className="collapse navbar-collapse"
                style={Object {}}
              >
                <Nav
                  className="mr-auto"
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <ul
                    className="mr-auto navbar-nav"
                  />
                </Nav>
                <p
                  className="user_name"
                >
                  <span
                    className="name"
                  >
                    Welcome, 
                    Keven
                  </span>
                   
                  <br />
                  <button
                    className="logout"
                    onClick={[Function]}
                    type="button"
                  >
                    Logout
                  </button>
                </p>
                <img
                  alt="user"
                  className="user_avatar"
                />
              </div>
            </Transition>
          </Collapse>
        </nav>
      </Navbar>
    </div>
  </Header>
</Provider>
`;
